src/amg/azext_amg/restore.py

```diff
@@ -81,38 +81,66 @@ def _restore_components(grafana_url, restore_functions, tmpdir, components, http
 
 
 # Restore dashboards
-def _create_dashboard(grafana_url, file_path, http_headers):
+def _load_and_create_dashboard(grafana_url, file_path, http_headers):
     with open(file_path, 'r', encoding="utf8") as f:
         data = f.read()
 
     content = json.loads(data)
     content['dashboard']['id'] = None
 
+    create_dashboard(grafana_url, content, http_headers, overwrite=True)
+
+
+def create_dashboard(grafana_url, content, http_headers, overwrite):
     payload = {
         'dashboard': content['dashboard'],
         'folderId': get_folder_id(content, grafana_url, http_post_headers=http_headers),
-        'overwrite': True
+        'overwrite': overwrite
     }
 
     datasources_missed = set()
     remap_datasource_uids(payload, uid_mapping, datasources_missed)
```

Comment:
I remember you said the change I made to the remap function was causing issues. Are you still experiencing the problem here?

Reply:
Yeah, both in migrate & restore it is broken, just re-tested again (Make AMG instance, create a datasource, then migrate, then the dashboard datasource uid isn't overwritten). I am not sure why my test didn't catch it though, going to double check.

Reply:
Okay fixed the test, now it is failing as expected:

https://github.com/Azure/azure-cli-extensions/pull/7802/commits/6efcf36e699007a365b5897f6f831c2b2c37935f

Reply:
Was weird edge case with test datasource, put in a read UID so now does remapping: [e23ecf8](https://github.com/Azure/azure-cli-extensions/pull/7802/commits/e23ecf80394275203417b81a8b1fd3600386d56c)